From d0cbf388919364fe6b5b9127f36426eb744090a0 Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston <jeremyhu@apple.com>
Date: Sun, 26 Sep 2010 21:21:16 -0700
Subject: [PATCH] Add an "X11_" string to header guards to avoid possible collision

This addresses a build failure which can result from <X11/Xlocale.h> and
<xlocale.h> being included in the same code since they both used the same
_XLOCALE_H_ protection.

Signed-off-by: Jeremy Huddleston <jeremyhu@apple.com>
---
 include/X11/ImUtil.h     |    6 +++---
 include/X11/XKBlib.h     |    6 +++---
 include/X11/Xcms.h       |    6 +++---
 include/X11/Xlib-xcb.h   |    6 +++---
 include/X11/Xlib.h       |    6 +++---
 include/X11/Xlibint.h    |    6 +++---
 include/X11/Xlocale.h    |    6 +++---
 include/X11/Xregion.h    |    6 +++---
 include/X11/Xresource.h  |    6 +++---
 include/X11/Xutil.h      |    6 +++---
 include/X11/cursorfont.h |    6 +++---
 11 files changed, 33 insertions(+), 33 deletions(-)

diff --git a/include/X11/ImUtil.h b/include/X11/ImUtil.h
index 5b62571..ffdba1a 100644
--- a/include/X11/ImUtil.h
+++ b/include/X11/ImUtil.h
@@ -1,6 +1,6 @@
 
-#ifndef _IMUTIL_H_
-#define _IMUTIL_H_
+#ifndef _X11_IMUTIL_H_
+#define _X11_IMUTIL_H_
 
 extern int
 _XGetScanlinePad(
@@ -27,4 +27,4 @@ extern void
 _XInitImageFuncPtrs(
     register XImage *image);
 
-#endif /* _IMUTIL_H_ */
+#endif /* _X11_IMUTIL_H_ */
diff --git a/include/X11/XKBlib.h b/include/X11/XKBlib.h
index 291fc05..8f6c72c 100644
--- a/include/X11/XKBlib.h
+++ b/include/X11/XKBlib.h
@@ -24,8 +24,8 @@ THE USE OR PERFORMANCE OF THIS SOFTWARE.
 
 ********************************************************/
 
-#ifndef _XKBLIB_H_
-#define _XKBLIB_H_
+#ifndef _X11_XKBLIB_H_
+#define _X11_XKBLIB_H_
 
 #include <X11/Xlib.h>
 #include <X11/extensions/XKBstr.h>
@@ -1146,4 +1146,4 @@ extern void XkbUpdateKeyTypeVirtualMods(
 
 _XFUNCPROTOEND
 
-#endif /* _XKBLIB_H_ */
+#endif /* _X11_XKBLIB_H_ */
diff --git a/include/X11/Xcms.h b/include/X11/Xcms.h
index cea11ab..8151fd1 100644
--- a/include/X11/Xcms.h
+++ b/include/X11/Xcms.h
@@ -26,8 +26,8 @@
  *	DESCRIPTION
  *		Public include file for X Color Management System
  */
-#ifndef _XCMS_H_
-#define _XCMS_H_
+#ifndef _X11_XCMS_H_
+#define _X11_XCMS_H_
 
 #include <X11/Xlib.h>
 
@@ -800,4 +800,4 @@ extern Visual *XcmsVisualOfCCC (
 
 _XFUNCPROTOEND
 
-#endif /* _XCMS_H_ */
+#endif /* _X11_XCMS_H_ */
diff --git a/include/X11/Xlib-xcb.h b/include/X11/Xlib-xcb.h
index 03d8f98..a21e2be 100644
--- a/include/X11/Xlib-xcb.h
+++ b/include/X11/Xlib-xcb.h
@@ -1,8 +1,8 @@
 /* Copyright (C) 2003-2006 Jamey Sharp, Josh Triplett
  * This file is licensed under the MIT license. See the file COPYING. */
 
-#ifndef XLIB_XCB_H
-#define XLIB_XCB_H
+#ifndef _X11_XLIB_XCB_H_
+#define _X11_XLIB_XCB_H_
 
 #include <xcb/xcb.h>
 #include <X11/Xlib.h>
@@ -17,4 +17,4 @@ void XSetEventQueueOwner(Display *dpy, enum XEventQueueOwner owner);
 
 _XFUNCPROTOEND
 
-#endif /* XLIB_XCB_H */
+#endif /* _X11_XLIB_XCB_H_ */
diff --git a/include/X11/Xlib.h b/include/X11/Xlib.h
index 7c8bc51..5c6c770 100644
--- a/include/X11/Xlib.h
+++ b/include/X11/Xlib.h
@@ -30,8 +30,8 @@ in this Software without prior written authorization from The Open Group.
  *	interface library (Xlib) to the X Window System Protocol (V11).
  *	Structures and symbols starting with "_" are private to the library.
  */
-#ifndef _XLIB_H_
-#define _XLIB_H_
+#ifndef _X11_XLIB_H_
+#define _X11_XLIB_H_
 
 #define XlibSpecificationRelease 6
 
@@ -4020,4 +4020,4 @@ extern void XFreeEventData(
 
 _XFUNCPROTOEND
 
-#endif /* _XLIB_H_ */
+#endif /* _X11_XLIB_H_ */
diff --git a/include/X11/Xlibint.h b/include/X11/Xlibint.h
index 77f0dc7..05d344c 100644
--- a/include/X11/Xlibint.h
+++ b/include/X11/Xlibint.h
@@ -27,8 +27,8 @@ from The Open Group.
 
 */
 
-#ifndef _XLIBINT_H_
-#define _XLIBINT_H_ 1
+#ifndef _X11_XLIBINT_H_
+#define _X11_XLIBINT_H_ 1
 
 /*
  *	Xlibint.h - Header definition and support file for the internal
@@ -1394,4 +1394,4 @@ extern void xlocaledir(
 
 _XFUNCPROTOEND
 
-#endif /* _XLIBINT_H_ */
+#endif /* _X11_XLIBINT_H_ */
diff --git a/include/X11/Xlocale.h b/include/X11/Xlocale.h
index baa1335..98946e5 100644
--- a/include/X11/Xlocale.h
+++ b/include/X11/Xlocale.h
@@ -26,8 +26,8 @@ from The Open Group.
 
 */
 
-#ifndef _XLOCALE_H_
-#define _XLOCALE_H_
+#ifndef _X11_XLOCALE_H_
+#define _X11_XLOCALE_H_
 
 #include <X11/Xfuncproto.h>
 #include <X11/Xosdefs.h>
@@ -56,4 +56,4 @@ _XFUNCPROTOEND
 
 #endif /* X_LOCALE */
 
-#endif /* _XLOCALE_H_ */
+#endif /* _X11_XLOCALE_H_ */
diff --git a/include/X11/Xregion.h b/include/X11/Xregion.h
index 7b376e5..c25d70b 100644
--- a/include/X11/Xregion.h
+++ b/include/X11/Xregion.h
@@ -45,8 +45,8 @@ SOFTWARE.
 
 ************************************************************************/
 
-#ifndef _XREGION_H
-#define _XREGION_H
+#ifndef _X11_XREGION_H_
+#define _X11_XREGION_H_
 
 typedef struct {
     short x1, x2, y1, y2;
@@ -186,4 +186,4 @@ typedef struct _POINTBLOCK {
     struct _POINTBLOCK *next;
 } POINTBLOCK;
 
-#endif
+#endif /* _X11_XREGION_H_ */
diff --git a/include/X11/Xresource.h b/include/X11/Xresource.h
index dbb144c..6dbb3ce 100644
--- a/include/X11/Xresource.h
+++ b/include/X11/Xresource.h
@@ -46,8 +46,8 @@ SOFTWARE.
 
 ******************************************************************/
 
-#ifndef _XRESOURCE_H_
-#define _XRESOURCE_H_
+#ifndef _X11_XRESOURCE_H_
+#define _X11_XRESOURCE_H_
 
 #ifndef _XP_PRINT_SERVER_
 #include <X11/Xlib.h>
@@ -354,5 +354,5 @@ extern void XrmParseCommand(
 
 _XFUNCPROTOEND
 
-#endif /* _XRESOURCE_H_ */
+#endif /* _X11_XRESOURCE_H_ */
 /* DON'T ADD STUFF AFTER THIS #endif */
diff --git a/include/X11/Xutil.h b/include/X11/Xutil.h
index 8ff34d9..702823f 100644
--- a/include/X11/Xutil.h
+++ b/include/X11/Xutil.h
@@ -46,8 +46,8 @@ SOFTWARE.
 
 ******************************************************************/
 
-#ifndef _XUTIL_H_
-#define _XUTIL_H_
+#ifndef _X11_XUTIL_H_
+#define _X11_XUTIL_H_
 
 /* You must include <X11/Xlib.h> before including this file */
 #include <X11/Xlib.h>
@@ -823,4 +823,4 @@ extern int XXorRegion(
 
 _XFUNCPROTOEND
 
-#endif /* _XUTIL_H_ */
+#endif /* _X11_XUTIL_H_ */
diff --git a/include/X11/cursorfont.h b/include/X11/cursorfont.h
index 70d7125..c69d508 100644
--- a/include/X11/cursorfont.h
+++ b/include/X11/cursorfont.h
@@ -26,8 +26,8 @@ from The Open Group.
 
 */
 
-#ifndef _cursorfont_h_
-#define _cursorfont_h_
+#ifndef _X11_CURSORFONT_H_
+#define _X11_CURSORFONT_H_
 
 #define XC_num_glyphs 154
 #define XC_X_cursor 0
@@ -108,4 +108,4 @@ from The Open Group.
 #define XC_watch 150
 #define XC_xterm 152
 
-#endif /* _cursorfont_h_ */
+#endif /* _X11_CURSORFONT_H_ */
-- 
1.7.2.3

